Aspekt.Bootstrap/IlGenerator.cs

```diff
@@ -167,38 +167,116 @@ public static void LoadArg(InstructionHelper ic, CustomAttributeArgument arg)
+        {
+            if (targetMethod.ReturnType.MetadataType == MetadataType.Void)
+            {
+                WeaverLog.LogMethodWarning(targetMethod, 1, "return value handler on void function");
+                return;
+            }
+
+            // Is derived from IAspectExitHandler<T>
+            var iExitHandler = methArgs.VariableType.Module.ImportReference(typeof(IAspectExitHandler<>));
+            var gii = new GenericInstanceType(iExitHandler);
+            gii.GenericArguments.Add(targetMethod.ReturnType);
+
+            var fullName = gii.ElementType.Resolve().FullName;
+
+            var attrInst = attrVar.VariableType.Resolve();
+
+            // If it implements the right interface
+            if (!attrInst.Interfaces.Any(i => i.InterfaceType.FullName == gii.FullName))
+            {
+                WeaverLog.LogMethodWarning(targetMethod, 2, $"return value handler implements invalid handler type; requires {gii.FullName}");
+                return;
+            }
+
+            // I just needed a type to resolve the name.
+            var methodName = nameof(IAspectExitHandler<int>.OnExit);
+            var method = attrInst.Methods.First(md => md.Name == methodName && md.Parameters.Count == 2 && md.Parameters[1].ParameterType.MetadataType == targetMethod.ReturnType.MetadataType);
+
+            // adjust all the return instructions
+            for (var i = 0; i < targetMethod.Body.Instructions.Count; ++i)
+            {
+                var inst = targetMethod.Body.Instructions[i];
+                // Find each return code, and add our three instructions.
+                if (inst.OpCode == OpCodes.Ret)
+                {
+                    var rep = inst;
+                    rep = inst.Previous; // get the previous instruction, because this SHOULD be the retval
+
+                    if (IsLoadInstruction(rep.OpCode))
+                    {
+                        InsertInstructionsAt(il, targetMethod, rep, il.Create(OpCodes.Ldloc, attrVar), il.Create(OpCodes.Ldloc, methArgs));
+                        i = i + 2;
+                    }
+                    else
+                    {
+                        // Create a variable, store it, load the
+                        var ih = new InstructionHelper(targetMethod.Module, il, rep, InstructionHelper.Insert.After);
+                        var retVar = ih.NewVariable(targetMethod.ReturnType);
+                        ih.Next(il.Create(OpCodes.Stloc_S, retVar));
+                        ih.Next(il.Create(OpCodes.Ldloc, attrVar));
+                        ih.Next(il.Create(OpCodes.Ldloc, methArgs));
+                        ih.Next(il.Create(OpCodes.Ldloc, retVar));
+                        i = i + 3;
+                    }
+
+                    // If it's a load instruction, I can just go one up, and put the   
```

Comment:
These comments seem a bit out of whack, reference the old method name and don't all make complete sentences.